color/src/dynamic.rs

```diff
@@ -365,28 +366,29 @@ impl DynamicColor {
     }
 }
 
-impl Hash for DynamicColor {
-    /// The hash is computed from the bit representation of the component values.
-    /// That makes it suitable for use as a cache key or memoization, but does not
-    /// match behavior for Rust float types.
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        self.cs.hash(state);
-        self.missing.hash(state);
-        for c in self.components {
-            c.to_bits().hash(state);
-        }
+impl PartialEq for DynamicColor {
+    /// Equality is not perceptual, but requires the component values to be equal.
+    ///
+    /// See also [`CacheKey`](crate::cache_key::CacheKey).
+    fn eq(&self, other: &Self) -> bool {
+        // Same as the derive implementation, but we want a doc comment.
+        self.cs == other.cs && self.missing == other.missing && self.components == other.components
     }
 }
 
-impl PartialEq for DynamicColor {
-    /// Equality is determined based on the bit representation.
-    fn eq(&self, other: &Self) -> bool {
+impl BitEq for DynamicColor {
+    fn bit_eq(&self, other: &Self) -> bool {
         self.cs == other.cs
             && self.missing == other.missing
-            && self.components[0].to_bits() == other.components[0].to_bits()
-            && self.components[1].to_bits() == other.components[1].to_bits()
-            && self.components[2].to_bits() == other.components[2].to_bits()
-            && self.components[3].to_bits() == other.components[3].to_bits()
+            && self.components.bit_eq(&other.components)
+    }
+}
```

Comment:
To be clear, they *probably* shouldn't cache the serialisation and instead just get it on-demand. But I guess if you're doing string interning, it might make sense?

Reply:
Yes, I think it makes sense. For example, when pretty-printing a style sheet, you may expect the same color to show up many times.